libass/ass.c

```diff
@@ -96,19 +96,20 @@ void ass_free_track(ASS_Track *track)
 
 /// \brief Allocate a new style struct
 /// \param track track
-/// \return style id
+/// \return style id or negative value on failure
 int ass_alloc_style(ASS_Track *track)
 {
     int sid;
 
     assert(track->n_styles <= track->max_styles);
 
     if (track->n_styles == track->max_styles) {
-        track->max_styles += ASS_STYLES_ALLOC;
-        track->styles =
-            (ASS_Style *) realloc(track->styles,
-                                  sizeof(ASS_Style) *
-                                  track->max_styles);
+        if (track->max_styles >= FFMIN(SIZE_MAX, INT_MAX) - ASS_STYLES_ALLOC)
```

Comment:
Minor thing, but: couldn’t this be `>` rather than `>=`, here and elsewhere? (I haven’t looked up what our existing overflow checks use.)

Reply:
When only integer types are involved, `>` would also be safe afaik. However the already existing limit check in ass.c uses `>=`: https://github.com/libass/libass/blob/152d0484e98f118d01987138695cf40579c9e297/libass/ass.c#L158-L160

Reply:
Also, I think we can safely assume `INT_MAX < SIZE_MAX`. We've already assumed 32-bit int elsewhere and libass cannot compile for 16-bit microcontrollers anyway (for example, default bitmap cache size is in order of ~100MB).